### PR TITLE
UXP docs

### PR DESCRIPTION
**Pull request by padmkris123**
Preparing documentation for public-beta release with UXP documentation

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/pages/plugins/concepts/entry-points/index.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  - https://github.com/padmkris123
+---
+
+# UXP Entry Points
+
+Plugins can have multiple entry points. An entrypoint is a way for a plugin to be invoked.
+
+There are two types of entry points: `commands` and `panels`. 
+
+- Commands are _"run and forget"_ entry points that are invoked by the user, such as a menu item or a keyboard shortcut.
```
Comment (padmkris123) [inline src/pages/plugins/concepts/entry-points/index.md:18]: @ ledzepfan Can you confirm if both types of plugins (command and panels) work in Pr?

File: src/pages/plugins/concepts/manifest/index.md [hunk 2]
```diff
@@ -1,5 +1,922 @@
...
+        <td>
+            <p>The maximum version of the host app that the plugin supports.</p>
+            <p><strong>Default value</strong></p>
+            <p><inlineCode>undefined</inlineCode> (the latest version of the host app)</p>
+        </td>
+    </tr>
+    </tbody>
+</table>
+
+### PermissionsDefinition
```
Comment (padmkris123) [inline src/pages/plugins/concepts/manifest/index.md:561]: @ ledzepfan Can you check if this list is up to date?

File: src/pages/plugins/getting-started/create-plugin.png
Comment (padmkris123) [inline src/pages/plugins/getting-started/create-plugin.png:1]: @ ledzepfan, Replace this file with only the modal dialog screenshot with the Host Application dropdown expanded.

File: src/pages/plugins/getting-started/loaded-plugin.png
Comment (padmkris123) [inline src/pages/plugins/getting-started/loaded-plugin.png:1]: Needs to be replaced with a screen shot of a generic plugin panel

File: src/pages/plugins/getting-started/load-plugin.png
Comment (padmkris123) [inline src/pages/plugins/getting-started/load-plugin.png:1]: Replace screenshot. Try to make it generic.

File: src/pages/plugins/getting-started/reload-plugin.png
Comment (padmkris123) [inline src/pages/plugins/getting-started/reload-plugin.png:1]: Replace screenshot. Try to make it generic

File: src/pages/plugins/tutorials/adding-command-entrypoints/command-plugin.png
Comment (padmkris123) [inline src/pages/plugins/tutorials/adding-command-entrypoints/command-plugin.png:1]: Replace with a generic screen shot

File: src/pages/plugins/tutorials/adding-command-entrypoints/index.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
+# Adding command entrypoints to your plugin
+
+Command entrypoints are the most common type of entrypoint. They allow you to add commands to the Premiere Pro user interface. When the user runs a command, your plugin can perform a variety of actions, such as opening a dialog and changing the document.
```
Comment (padmkris123) [inline src/pages/plugins/tutorials/adding-command-entrypoints/index.md:3]: Test if this works in Pr.

File: src/pages/plugins/tutorials/importing-modules/index.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# JavaScript Modules
+
+The beauty of plugins is that it allows you to have more than just one JavaScript file. And in order to manage elaborate projects, you might want to modularize your code and manage it from separate JS files.
```
Comment (padmkris123) [inline src/pages/plugins/tutorials/importing-modules/index.md:3]: Test if this works

File: src/pages/plugins/tutorials/inter-plugin-comm/index.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
+---
+title: Inter Plugin Communication
```
Comment (padmkris123) [inline src/pages/plugins/tutorials/inter-plugin-comm/index.md:2]: Test if this works

File: src/pages/plugins/tutorials/plugin-lifecycle-hooks/index.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+description: Use plugin life cycle events to carry out load/unload routines
+keywords:
+  - plugin hooks
+  - lifecycle
+contributors:
+  - https://github.com/padmkris123
+---
+
+
+# Plugin Lifecycle events
```
Comment (padmkris123) [inline src/pages/plugins/tutorials/plugin-lifecycle-hooks/index.md:12]: Test if this works

File: src/pages/plugins/tutorials/udt-deep-dive/images/create-plugin.png
Comment (padmkris123) [inline src/pages/plugins/tutorials/udt-deep-dive/images/create-plugin.png:1]: Replace with generic screenshot

File: src/pages/resources/fundamentals/create-ui/index.md [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+  - Spectrum in UXP
+  - UI
+  - User interface
+title: Create UI in UXP Spectrum UXP Reference
+description: Describes the options available within UXP to create UI
+contributors:
+  - https://github.com/padmkris123
+---
+
+# User Interface (UI)
```
Comment (padmkris123) [inline src/pages/resources/fundamentals/create-ui/index.md:17]: Anything you would like to word differently here?

File: src/pages/resources/recipes/clipboard/index.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+title: Clipboard
+description: Read and write content from and to the clipboard.
+keywords:
+  - Read write
+  - Clipboard sample
+contributors:
+  - https://github.com/padmkris123
+---
+
+# Clipboard
```
Comment (padmkris123) [inline src/pages/resources/recipes/clipboard/index.md:11]: Test

File: src/pages/resources/recipes/css-styling/index.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
+---
+title: CSS styling
+description: Techniques to style UI in UXP
+keywords:
+  - CSS in UXP
+contributors:
+  - https://github.com/padmkris123
+---
+
+# CSS styling
```
Comment (padmkris123) [inline src/pages/resources/recipes/css-styling/index.md:10]: Test

File: src/pages/resources/recipes/file-operation/index.md [hunk 10]
```diff
@@ -0,0 +1,330 @@
...
+description: Perform file operations in UXP
+keywords:
+  - localFileSystem
+  - FS API
+  - fullaccess
+contributors:
+  - https://github.com/padmkris123
+---
+
+# File operations
```
Comment (padmkris123) [inline src/pages/resources/recipes/file-operation/index.md:12]: Test

File: src/pages/resources/recipes/host-info/index.md [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+description: Find the host info to determine locale, os etc
+keywords:
+  - Host API
+  - code sample 
+  - locale
+contributors:
+  - https://github.com/padmkris123
+---
+
+# Host Environment
```
Comment (padmkris123) [inline src/pages/resources/recipes/host-info/index.md:12]: Test

File: src/pages/resources/recipes/html-elements/index.md [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+title: HTML elements
+description: Ways to create HTML elements
+keywords:
+  - User interface
+  - UI in UXP
+contributors:
+  - https://github.com/padmkris123
+---
+
+# HTML Elements
```
Comment (padmkris123) [inline src/pages/resources/recipes/html-elements/index.md:11]: Test

File: src/pages/resources/recipes/html-events/index.md [hunk 13]
```diff
@@ -0,0 +1,89 @@
...
+description: Ways to configure HTML events
+keywords:
+  - event listeners
+  - sample html events
+contributors:
+  - https://github.com/padmkris123
+---
+
+
+# HTML events and event listeners
```
Comment (padmkris123) [inline src/pages/resources/recipes/html-events/index.md:12]: Test

File: src/pages/resources/recipes/launch-process/index.md [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+description: Launch applications and files in UXP
+keywords:
+  - shell
+  - external process
+contributors:
+  - https://github.com/padmkris123
+---
+
+
+# Launch process
```
Comment (padmkris123) [inline src/pages/resources/recipes/launch-process/index.md:12]: Test

File: src/pages/resources/recipes/network/index.md [hunk 15]
```diff
@@ -0,0 +1,191 @@
...
+description: Network pi
+keywords:
+  - fetch
+  - xhr
+  - websockets 
+contributors:
+  - https://github.com/padmkris123
+---
+
+# Network I/O
```
Comment (padmkris123) [inline src/pages/resources/recipes/network/index.md:12]: Test

File: src/pages/uxp-api/known-issues.md [hunk 16]
```diff
@@ -0,0 +1,107 @@
+<!--
+index_desc: List of known issues in UXP
+-->
+
+# Known Issues
```
Comment (padmkris123) [inline src/pages/uxp-api/known-issues.md:5]: Do all these apply for Pr? Do we need to add anything else?

File: src/pages/plugins/concepts/entry-points/index.md (hunk 1 above)
Comment (ledzepFan) [reply src/pages/plugins/concepts/entry-points/index.md]: Yes

File: src/pages/resources/recipes/flyout-menu/index.md [hunk 17]
```diff
@@ -0,0 +1,83 @@
+---
+keywords:
+  - API Documentation
+  - UXP
+  - Flyout Menu
+  - Hamburger Menu
+title: Flyout Menus in Premiere Pro UXP plugins
+---
+
+# Flyout Menus
```
[the remark this answers is not in the document]
Comment (padmkris123) [reply src/pages/resources/recipes/flyout-menu/index.md]: Removed

File: src/pages/resources/fundamentals/create-ui/index.md (hunk 7 above)
Comment (padmkris123) [reply src/pages/resources/fundamentals/create-ui/index.md]: Need to revisit for GA

**Pull request merged**